xbmc/guilib/GUIAction.cpp

```diff
@@ -72,23 +98,43 @@ void CGUIAction::SetNavigation(int id)
   std::string strId = std::to_string(id);
   for (auto &i : m_actions)
   {
-    if (StringUtils::IsInteger(i.action) && i.condition.empty())
+    if (StringUtils::IsInteger(i.GetAction()) && !i.HasCondition())
     {
-      i.action = std::move(strId);
+      i.SetAction(std::move(strId));
       return;
     }
   }
   m_actions.emplace_back();
-  m_actions.back().action = std::move(strId);
+  m_actions.back().SetAction(std::move(strId));
```

Comment:
just do `m_actions.emplace_back(strId, std::string());` then we don't need the default ctor

Reply:
Imo, this std::move buzz is nonsense for the use cases we use it. Are those function really performance/ressource saving critical? Just as a side note.